ClothesApp/Infrastructure/Services/BrandService.cs

```diff
@@ -0,0 +1,100 @@
+using Application.Dtos;
+using Application.Repositories;
+using Application.Services;
+using AutoMapper;
+using Domain.Entities;
+using Infrastructure.Exceptions;
+using Microsoft.EntityFrameworkCore;
+
+namespace Infrastructure.Services;
+
+public class BrandService : IBrandService
+{
+    private readonly IBrandsRepository _brandsRepository;
+
+    private readonly IProductRepository _productRepository;
+
+    private readonly IMapper _mapper;
+
+    public BrandService(IBrandsRepository brandsRepository, IProductRepository productRepository, IMapper mapper)
+    {
+        _brandsRepository = brandsRepository;
+        _productRepository = productRepository;
+        _mapper = mapper;
+    }
+
+    public async Task<BrandDto> AddBrandAsync(CreateBrandDto createBrandDto)
+    {
+        var brandExists = await _brandsRepository.IsExistAsync(createBrandDto.Name);
+
+        if (brandExists)
+        {
+            throw new DuplicationException();
```

Comment:
For client there is no difference between RelationExistException and DuplicationException, both have Conflict as a status code with different string, I would suggest you to think more about exceptions model because you will be very busy with creating a lot of new exception types that have same status codes and handlers, if the only difference is exception message, then you can pass it right here